play-json-ops-common/src/main/scala/play/api/libs/json/ops/TolerantContainerFormats.scala

```diff
@@ -0,0 +1,24 @@
+package play.api.libs.json.ops
+
+import play.api.libs.json.{JsPath, Reads}
+
+import scala.collection.generic
+import scala.language.{higherKinds, implicitConversions}
+
+/**
+ * Provides helpers which will create empty containers if a json field is missing
+ *
+ * @see [[play.api.libs.json.ops.TolerantContainerPath.readNullableContainer]]
+ */
+trait TolerantContainerFormats {
+  implicit final def getTolerantContainerPath(jsPath: JsPath): TolerantContainerPath = new TolerantContainerPath(jsPath)
+}
+
+class TolerantContainerPath(val jsPath: JsPath) extends AnyVal {
+
+  /**
+   * Defines a [[Reads]] that will read all container fields such that `null=[]` (or an empty [[Set]] / [[Map]])
+   */
+  def readNullableContainer[C[_], A](implicit r: Reads[C[A]], bf: generic.CanBuildFrom[C[_], A, C[A]]): Reads[C[A]] =
+    Reads.nullable[C[A]](jsPath)(r).map(_.getOrElse(bf().result()))
```

Comment:
As I remember this will not work at the scala 2.13. Since `CanBuildFrom` is changed to `BuildFrom` and it has an argument `from`.

Reply:
Scala 2.13 has its own source code because of this (and other reasons)